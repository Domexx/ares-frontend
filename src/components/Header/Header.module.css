/* === === === === Header === === === === */

.Header {
  position: relative;
  overflow: hidden;
  height: 150px;
  border-bottom: 3px solid #0808082e;
  text-align: center;
  text-shadow: #0000004d 0 1px 0;
  background: url('/static/images/citysky.png') repeat-x, url('/static/images/cloudysky.png') rgb(116, 216, 234);
  background-repeat: repeat-x;
}

.Header::before {
  content: '';
  position: absolute;
  display: block;
  background-color: #00000069;
  background-position: center bottom;
  inset: 0;
}

.Header:global {
  animation: 60s linear 0s infinite normal none running slide;
}
