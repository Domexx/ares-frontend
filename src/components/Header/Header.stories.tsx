import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import HeaderComponent, { Props } from './Header';

const Component = {
  title: 'Components/Header',
  component: HeaderComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof HeaderComponent>> = (args) => {
  return <HeaderComponent {...args} />;
};

export const Header = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
Header.args = {
  text: 'Hello',
} as Props;
