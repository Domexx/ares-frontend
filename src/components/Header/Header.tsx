import React from 'react';
import s from './Header.module.css';
import Logo from '../Logo';

export type Props = {
  text: string;
};

const Header: React.FC<Props> = ({ text }) => {
  return (
    <div className={s['Header']}>
      <div className="container h-full items-center">{text}</div>
      <div className={s['image']}>
        <Logo />
      </div>
    </div>
  );
};

export default Header;
