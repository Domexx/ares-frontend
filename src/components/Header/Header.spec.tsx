import React from "react";

import { render } from "@testing-library/react";

import Header from "./Header";

describe("Header Component", () => {

  it("should render properly", () => {
    const { container } = render(
      <Header text="Test Text" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});