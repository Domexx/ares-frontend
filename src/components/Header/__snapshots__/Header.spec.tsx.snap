// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render properly 1`] = `
<div
  class="Header"
>
  <div
    class="container h-full items-center"
  >
    Test Text
  </div>
  <div
    class="image"
  />
</div>
`;
